src/Common/ThreadFuzzer.cpp

```diff
@@ -159,13 +162,31 @@ bool ThreadFuzzer::isEffective() const
             || (sleep_probability > 0 && sleep_time_us > 0));
 }
 
+void ThreadFuzzer::stopFuzzing()
+{
+    stop_fuzzing.store(true);
```

Comment:
Put memory_order_relaxed everywhere. Otherwise it will slow down every `injection` function even if ThreadFuzzer is disabled.